src/teleop/gui/src/components/Cameras.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+    <div class="wrap">
+      <h3>Cameras</h3>
+      <div class="input">
+        Camera name: <input class="rounded" type='message' v-model ='cameraName'>
+        Camera number: <input class="rounded" type='Number' min="0" max="9" v-model ='cameraIdx'>
+        <button class="rounded button" v-on:click="addCameraName()">Change name</button>
+      </div>
+      <div class="cameraselection">
+        <CameraSelection class="cameraspace1" v-bind:camsEnabled="camsEnabled" v-bind:names="names" v-bind:capacity="parseInt(capacity)" v-on:cam_index="setCamIndex($event)"/>
+      </div>
+      <h3>All Cameras</h3>
+      Capacity: <input class="rounded" type='Number' min="2" max="4" v-model ='capacity'>
+      <div class="camerainfo" v-for="i in camsEnabled.length" :key="i">
+        <CameraInfo v-if="camsEnabled[i-1]" v-bind:name="names[i-1]" v-bind:id="i-1" v-on:newQuality="changeQuality($event)" v-on:swapStream="swapStream($event)" v-bind:stream="getStreamNum(i-1)"></CameraInfo>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  import ROSLIB from 'roslib/src/RosLib'
+  import CameraSelection from '../components/CameraSelection.vue'
+  import CameraInfo from '../components/CameraInfo.vue'
+  
+  export default {
+    data() {
+      return {
+        camsEnabled: new Array(9).fill(false),
+        names: Array.from({length: 9}, (_,i) => "Camera: " + i),
+        cameraIdx: 1,
+        cameraName: "",
+        capacity: 2,
+        qualities: new Array(9).fill(1),
+        streamOrder: [-1, -1, -1, -1]
+      }
+    },
+
+    props: {
+      numCams: {
+        type: Number,
+        required: true
+      },
+      mission: {
+        type: String,
+        required: true
+      },
+      channel: {
+        type: String,
+        required: true
+      },
+      primary: {
+        type: Boolean,
+        required: true
+      }
+
+    },
+
+
+    methods: {
+
+      setCamIndex: function (index) { //every time a button is pressed, it changes cam status and adds/removes from stream
+        this.camsEnabled.splice(index, 1, !this.camsEnabled[index]);
+        this.changeStream(index);
+        this.sendCameras();
+      },
+
+      sendCameras: function() { //sends cameras to a service to display on screen
+        var msgs = [];
+        for(var i = 0; i < 4; i++){
+          var camId = this.streamOrder[i];
+          var res = this.qualities[camId];
+          msgs.push(new ROSLIB.Message({device: camId, resolution: res})); //CameraCmd msg
+        }
+
+        var changeCamsService = new ROSLIB.Service({
+              ros : this.$ros,
+              name : 'change_cameras',
+              serviceType : 'ChangeCameras'
```

Comment:
has this been tested? I don't see the [ChangeCameras.srv](https://github.com/umrover/mrover-ros/blob/guts/cameras/srv/ChangeCameras.srv) and [CameraCmd.msg](https://github.com/umrover/mrover-ros/blob/guts/cameras/msg/CameraCmd.msg) files in this repo so if i run this im not sure if it would work

Reply:
ok actually i don't think you can test this on this repo since there's no camera service server running rip... ok i'm not sure what the best plan of testing this entire merge is but there's two things we can do: 1. have a camera integration branch or 2. just merge this in now and then resolve changes later. I think method 1 is cleaner but 2 will prob make you happier lol